TVAULT-4840 Delete snapshot physically from database

nikhiljathartrilio opened the pull request:
@DivyaDas-Trilio @DeepikaAFour @dhiraj-trilio

Commit:
Delete snapshot physically from database

DeepikaAFour commented:
@nikhiljathartrilio 
1. Please mention in the documentation that only soft deleted workloads and snapshots will be deleted
2. Please mention what kind of negative testing have been done
3. Have you verified that all dependent records from other tables are cleaned or not?

Commit:
TVAULT-4840 Changes made in Readme file

nikhiljathartrilio commented:
@DeepikaAFour
1) I have updated ReadMe.md.
2) Following are some negative test's i have done.
  1) With empty ID file's.
  2) With available and executing workload and snapshot ID's.
  3) With wrong ID's.
  4) ID's which is not soft deleted.
3) yes, i have checked all dependent tables, entries in all dependent table also deleted if you try to delete workload or snapshot which already soft deleted in database.

dhiraj-trilio commented on openstack/CleanWlmDatabase/snapshot_delete.py:
```diff
@@ -0,0 +1,55 @@
+def delete_snapshots(curr, conn, snapshots_ids):
+    for snapshot_id in snapshot_ids:
+        try:
+            snapshot_exists = "SELECT EXISTS(SELECT * FROM snapshots WHERE id = %s AND deleted = 1 AND status = \"deleted\")"
```
can't we fetch all the given snaps at once, something like SELECT * FROM snapshots WHERE id in (<list_of_snap_ids>) AND deleted = 1 AND status = \"deleted\"

dhiraj-trilio commented on openstack/CleanWlmDatabase/snapshot_delete.py:
```diff
@@ -0,0 +1,55 @@
+    try:
+        conn, cursor = connect_db(Log_wlm)
+        if cursor:
+            snapshot_ids = file_manager('config/snapshot_ids.txt', Log_wlm, file_type='text', mode='r')
```
it could have been better to use argparser and take this config file from user as an argument instead of hard coding it.

dhiraj-trilio commented on openstack/CleanWlmDatabase/workloads_delete.py:
```diff
@@ -9,7 +9,7 @@
     for workload_id in workload_ids:
         try:
-            workload_exists = "SELECT EXISTS(SELECT * FROM workloads WHERE id = %s)"
+            workload_exists = "SELECT EXISTS(SELECT * FROM workloads WHERE id = %s AND deleted = 1 AND status = \"deleted\")"
```
can we fetch all the workloads at once ? and process further ?
